packages/interface/src/server/api/routers/results.ts

```diff
@@ -58,8 +58,11 @@ export async function calculateMaciResults(
     });
   }
 
-  const results = tallyData.results.reduce((acc, tally, index) => {
-    const project = projects[index];
+  const results = tallyData.results.reduce((acc, tally) => {
+    /// The tallyResult.id on old version subgraph is 0, 1, 2, ..., and is {tallyAddress}-{id=0, 1, 2, ...} on new version
+    const [id1, id2] = tally.id.split("-");
+    const projectIndex = id2 ? Number(id2) : Number(id1);
```

Comment:
This is to support old and new versions, right? shouldn't we only support new versions as we create the subgraph?

Reply:
I am not sure about this part, how do you think? if we only wanna support the new version, I'll delete this part :)

Reply:
mmm maybe let's support both? if someone spins up another round and wants to host both (or more) rounds and update the frontend/subgraph then one will work and the others not

Reply:
ok, then I'll keep this part

Reply:
Yeah let's keep it and once we release a new version v3 or something we ca remove it.